### PR TITLE
Feature: Add on private key encryption

### PR DESCRIPTION
**Pull request by SangTran-127**
## Type

- [x] Feature
- [ ] Bug fix
- [ ] CI/CD
- [ ] Documentation
- [ ] Other

## Description

Brief description of the changes made.

### CODE REVIEW

File: packages/proof-service/src/service/task-service.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ class TaskService {
     let retries = 0;
     let delay = this.initialDelay;
 
-    while (retries < this.maxRetries) {
+    while (true) {
```
Comment (magestrio) [inline packages/proof-service/src/service/task-service.ts:22]: Will k8s manager remove app process if we are in infinite loop?

File: packages/compile-service/src/service/zk-compile.ts [hunk 2]
```diff
@@ -32,26 +33,42 @@ export class ZkCompileService {
...
+      logger.info(
+        `Deploy ${zkDbPublicKey.toBase58()} take ${(end - start) / 1000}s`
+      );
+
+      return unsignedTx.toJSON();
+    } catch (error) {
+      logger.error(`Cannot compile & deploy: ${databaseName}`, logger);
+      await ModelDbSetting.getInstance().updateSetting(databaseName, {
+        appPublicKey: undefined,
+        deployStatus: "failed",
```
Comment (magestrio) [inline packages/compile-service/src/service/zk-compile.ts:68]: I think, here we should have 2 states: `deployed` | `not_deployed`. Maybe we should store the transaction regardless of success/failed and store the failed status there alongside with errorMessage

File: packages/proof-service/src/service/task-service.ts (hunk 1 above)
Comment (SangTran-127) [reply packages/proof-service/src/service/task-service.ts:22]: I'm forget to undo it. My bad. Just config while true to run on local machine

File: packages/compile-service/src/service/zk-compile.ts (hunk 2 above)
Comment (SangTran-127) [reply packages/compile-service/src/service/zk-compile.ts]: resolved

**Pull request merged**